### PR TITLE
Multi stage dockerfile

### PR DESCRIPTION
**Pull request by sjmiller609**
This is targeting remove_vendor until that one is merged to avoid duplicate diff. This is 2/2 in observance of https://github.com/astronomer/commander/pull/14#issuecomment-541156307

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,33 +2,33 @@
...
-		protobuf \
-		build-base \
-		ca-certificates \
-		go \
-		python2 \
-	&& mkdir -p "${GOPATH}" \
-	&& mkdir "${GOPATH}/src" \
-	&& mkdir "${GOPATH}/bin"
+RUN apt-get update && \
+    apt-get install -y dtrx
```
Comment (andriisoldatenko) [inline Dockerfile]: are you sure that we need dtrx? why not just unzip?

File: Dockerfile [hunk 2]
```diff
@@ -41,16 +41,29 @@ RUN go get -u github.com/golang/dep/cmd/dep
...
+
+COPY . ./
+
+# Build commander
 RUN make build-proto
-# Compile
-RUN make build
+RUN make build && cp ./commander /commander
+
+FROM alpine:latest
```
Comment (andriisoldatenko) [inline Dockerfile]: for security better to use alpine:3.10 (fixed version)

File: Makefile [hunk 3]
```diff
@@ -11,7 +11,10 @@ dependencies:
 	dep ensure -vendor-only -v
 
 build:
-	go build -o ${OUTPUT} main.go
+	CGO_ENABLED=0 go build -o ${OUTPUT} main.go
+
+dependencies:
+	dep ensure -vendor-only -v
```
Comment (andriisoldatenko) [inline Makefile]: duplicate line 10?

File: Dockerfile (hunk 1 above)
Comment (sjmiller609) [reply Dockerfile]: does not matter because multi-stage docker build?

File: Dockerfile (hunk 2 above)
Comment (sjmiller609) [reply Dockerfile]: ok

**Comment by sjmiller609** [comment]
@andriisoldatenko anything else?

File: Dockerfile [hunk 4]
```diff
@@ -2,33 +2,33 @@
...
-		python2 \
-	&& mkdir -p "${GOPATH}" \
-	&& mkdir "${GOPATH}/src" \
-	&& mkdir "${GOPATH}/bin"
+RUN apt-get update && \
+    apt-get install -y unzip
+
+RUN wget $PROTOC_URL \
+    && unzip ./${PROTOC_FILE}.zip \
+    && mv ./bin/protoc /usr/local/sbin/protoc
```
Comment (andriisoldatenko) [inline Dockerfile]: FYI you can do in 1 step: `unzip ./${PROTOC_FILE}.zip -d /usr/local/sbin/protoc`

Comment (sjmiller609) [reply Dockerfile]: wouldn't that put protoc like this?

/usr/local/sbin/bin/protoc
but I want
/usr/local/sbin/protoc

File: Dockerfile [hunk 5]
```diff
@@ -2,33 +2,33 @@
...
+
 MAINTAINER Astronomer <humans@astronomer.io>
 
 ARG BUILD_NUMBER=-1
 LABEL io.astronomer.docker.build.number=$BUILD_NUMBER
 LABEL io.astronomer.docker.module="astronomer"
-LABEL io.astronomer.docker.component="houston"
-LABEL io.astronomer.docker.environment="development"
+LABEL io.astronomer.docker.component="commander"
+LABEL io.astronomer.docker.environment="development-build"
```
Comment (schnie) [inline Dockerfile]: @sjmiller609 should we just remove the `LABEL`s up here and only keep them on the final image? Seems like `BUILD_NUMBER` should also be moved to the final image, not the build image.

**Comment by sjmiller609** [comment]
moving to master branch PR

**Pull request closed without merging**